packages/superset-ui-legacy-preset-chart-nvd3/src/NVD3Vis.js

```diff
@@ -254,6 +256,7 @@ function nvd3Vis(element, props) {
   const container = element;
   container.innerHTML = '';
   const activeAnnotationLayers = annotationLayers.filter(layer => layer.show);
+  const chartId = container.parentElement.id || null;
```

Comment:
perhaps also check if `parentElement` is not null.
Also, do we want to prevent accidents if `id` is `0` which is falsy?

Reply:
updated